spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -0,0 +1,679 @@
+            !.blocksRequested = pState.blocksRequested \ {msg}      
+        ] IN
+    [msg |-> m, peers |-> npState]   
+         
+GrowBlockchain(pState) == 
+    LET newPeerState == 
+        [pState EXCEPT !.peerHeights = [p \in AllPeerIds |-> pState.peerHeights[p] + 1]] 
+    IN
+    /\ peersState' = newPeerState
+    /\ inMsg' = NoMsg        
+
+
+SendStatusResponseMessage(pState) == 
+    /\ \E h \in Heights: 
+        \E peer \in AllPeerIds:   
+            LET msgAndPeers == CreateStatusResponse(peer, pState, h) IN
+               /\ peersState' = msgAndPeers.peers
+               /\ inMsg' = msgAndPeers.msg   
+
+
+SendAddPeerMessage == 
+   \E peer \in AllPeerIds:
+     /\ inMsg' = [type |-> "addPeer", peerId |-> peer]
+     /\ UNCHANGED peersState    
+
+SendRemovePeerMessage == 
+   \E peer \in AllPeerIds:
+     /\ inMsg' = [type |-> "removePeer", peerId |-> peer]
+     /\ UNCHANGED peersState    
+
+SendSyncTimeoutMessage == 
+    /\ inMsg' = [type |-> "syncTimeout"]
+    /\ UNCHANGED peersState    
+
+
+SendControlMessage(pState) == 
+    \/ SendStatusResponseMessage(pState)
+    \/ SendAddPeerMessage
+    \/ SendRemovePeerMessage
+    \/ SendSyncTimeoutMessage
+        
+
+SendBlockResponseMessage(pState) == 
+    \/  /\ pState.blocksRequested /= {}
+        /\ \E msg \in pState.blocksRequested:
+             \E block \in Blocks:
+                LET msgAndPeers == CreateBlockResponse(msg, pState, block) IN      
+                 /\ peersState' = msgAndPeers.peers
+                 /\ inMsg' = msgAndPeers.msg      
+    
+    
+    \/  /\ peersState' = pState
+        /\ inMsg' \in [type: {"blockResponse"}, peerId: FAULTY, block: Blocks]    
+
+CreateResponse(pState) == 
+    \/  SendBlockResponseMessage(pState)
+    \/  GrowBlockchain(pState)
+    \/  SendControlMessage(pState)
+                         
+    
+\* Peers consume a message and update it's local state. It then makes a single step, i.e., it sends at most single message.
+\* Message sent could be either a response to a request or faulty message (sent by faulty processes). 
+NextPeers ==
+    LET pState == HandleRequest(outMsg, peersState) IN
+    
+    \/  /\ outMsg' = NoMsg                    \* correct peers respond
+        /\ CreateResponse(pState) 
+    
+
+\* the composition of the node, the peers, the network and scheduler
+Init ==
+    /\ InitNode
+    /\ InitPeers
+    /\ turn = "Peers"
+    /\ inMsg = NoMsg
+    /\ outMsg = [type |-> "statusRequest", peerIds |-> blockPool.peerIds]
+    /\ nonDetFlag \in BOOLEAN
+        
+Next ==
+    IF turn = "Peers"
+    THEN
+        /\ NextPeers
+        /\ turn' = "Node"
+        /\ nonDetFlag' \in BOOLEAN
+        /\ UNCHANGED nvars
+    ELSE
+        /\ NextNode
+        /\ turn' = "Peers"
+        /\ nonDetFlag' \in BOOLEAN
+        /\ UNCHANGED peersState  
+
+
+
+FlipTurn == 
+ turn' = (
+  IF turn = "Peers" THEN 
+   "Node" 
+  ELSE
+   "Peers"
+ ) 
+        
+ 
+
+\* properties to check
+TypeOK ==
```

Comment:
this also fails for me in initial state.

Reply:
The same for me. Not able to find a bug. I am still having few issues with verifying properties that I am trying to figure out in parallel :)

Reply:
Should be fine now.